docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc

```diff
@@ -1,33 +1,305 @@
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Get last value of a Data Packet
+auto lastValue = packet.getLastValue();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#printing_the_last_value]
+== Printing the last value
+
+No matter the source (either the Signal or the Data Packet), the resulting variable can be, for example, printed into the console.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Print last value
+std::cout << "lastValue: " << lastValue << std::endl;
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#casting_the_last_value]
+== Casting the last value and using the cast variable
+
+In C++, the function `getLastValue` returns an object of type `IBaseObject`. To access it's underying members and methods, one must explicitly cast it into the appropriate type. However, in Python, this is not necessary due to dynamic typing.
+
+The underlying type of the `IBaseObject` we get from our `getLastValue` call can be cast into either:
+
+* `IFloat` if the type is `Float32` or `Float64`
+* `IInteger` if the type is `Int8` through `Int64` or `UInt8` through `UInt64`
+* `IComplexNumber` if the type is `ComplexFloat32` or `ComplexFloat64`
+* `IRange` if the type is `RangeInt64`
+* `IStruct` if the type is `Struct`
+* `IList` of the aforementioned types if there is exactly one dimension in the single value
+
+An example of casting the `lastValue` object into an `IFloat` object:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IFloat
+auto number = lastValue.asPtr<IFloat>();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which `getLastValue` (`last_value` in Python) is called might be:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// The Data Descriptor for Float64
+auto descriptor = DataDescriptorBuilder().setSampleType(SampleType::Float64).build();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#last_value_range]
+== Getting the last value with a Range
+
+The following example demonstrates casting the `lastValue` object into an `IRange`. Something usefeul we can then do is call the methods of the underlying type and access its members, such as the `getLowValue` (`low_value` in Python) and the `getHighValue` (`high_value` in Python), in this case provided by the `IRange` interface.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IRange
+auto range = lastValue.asPtr<IRange>();
+// Extract values
+auto low = range.getLowValue();
+auto high = range.getHighValue();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which `getLastValue` (`last_value` in Python) is called is similar to the the one from the previous example (the Data Descriptor with Sample Type `Float64`), except the Sample Type is `RangeInt64`.
+
+[#last_value_complex_number]
+== Getting the last value with a Complex Number
```

Comment:
As above, I don't think an explanation of how complex numbers work is needed in this how-to